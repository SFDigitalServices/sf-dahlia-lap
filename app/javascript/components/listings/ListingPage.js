import React from 'react'

import ListingDetails from './ListingDetails'
import CardLayout from '../layouts/CardLayout'

const ListingPageDetails = ({ listing }) => {
  return <ListingDetails listing={listing} />
}

const ListingPage = ({ listing }) => {
  const tabs = {
    items: [
      { title: 'Listing Details', url: `/listings/${listing.id}`, active: true },
      { title: 'Applications', url: `/listings/${listing.id}/applications` }
    ]
  }

  return (
    <CardLayout pageHeader={{title: listing.name}} tabSection={tabs}>
      <ListingPageDetails listing={listing} />
    </CardLayout>
  )
}

export default ListingPage
