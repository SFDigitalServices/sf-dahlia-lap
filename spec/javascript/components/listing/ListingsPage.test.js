import React from 'react'
import renderer from 'react-test-renderer';

import ListingPage from 'components/listings/ListingPage'
import modelsFactory from '../../factories/models'
import sharedHooks from '../../support/sharedHooks'

describe('ListingsPage', () => {
  sharedHooks.useFakeTimers()
  test('should render succesfully', () => {
    const listing = modelsFactory.listing(1)

    const wrapper = renderer.create(
      <ListingPage listing={listing}/>,
    )

    expect(wrapper.toJSON()).toMatchSnapshot();
  })
})
