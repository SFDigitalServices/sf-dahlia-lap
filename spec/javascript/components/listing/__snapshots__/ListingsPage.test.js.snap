// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingsPage should render succesfully 1`] = `
Array [
  <header
    className="lead-header short  bg-snow"
  >
    <div
      className="row full-width inner--3x"
    >
      <div
        className="large-12 columns"
      >
        <hgroup
          className="lead-header_group"
        >
          <h1
            className="lead-header_title small-serif c-oil"
          >
            xxx2
          </h1>
        </hgroup>
      </div>
    </div>
  </header>,
  <div
    className="tabs-section tabs-row row full-width inner--3x bg-snow"
  >
    <ul
      className="tabs full-width-small-only"
      role="menubar"
    >
      <li
        className="tab-title"
        role="none"
      >
        <a
          aria-selected={false}
          href="/listings/1"
          onFocus={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="0"
        >
          Listing Details
        </a>
      </li>
      <li
        className="tab-title"
        role="none"
      >
        <a
          aria-selected={false}
          href="/listings/1/applications"
          onFocus={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="0"
        >
          Applications
        </a>
      </li>
    </ul>
  </div>,
  <section
    className="tabs-card-row row full-width inner--3x margin-bottom--2x"
  >
    <div
      className="app-card form-card tabs-card max-width expand-on-small"
    >
      <div>
        <div
          className="content-card padding-bottom-none margin-bottom--half bg-trans"
        >
          <h4
            className="content-card_title t-serif"
          >
            Details
          </h4>
          <ul
            className="content-grid"
          >
            <li
              className="content-item"
            >
              <div
                className="content-card"
              >
                <div
                  className="margin-bottom--half"
                >
                  <h4
                    className="t-sans t-small t-bold no-margin"
                  >
                    Name
                  </h4>
                  <p>
                    xxx2
                  </p>
                </div>
                <div
                  className="margin-bottom--half"
                >
                  <h4
                    className="t-sans t-small t-bold no-margin"
                  >
                    Application Due Date
                  </h4>
                  <p>
                    04/23/2018
                  </p>
                </div>
              </div>
            </li>
            <li
              className="content-item"
            >
              <div
                className="content-card"
              />
            </li>
          </ul>
        </div>
        <div
          className="content-card padding-bottom-none margin-bottom--half bg-trans"
        >
          <h4
            className="content-card_title t-serif"
          >
            Building Information
          </h4>
          <ul
            className="content-grid"
          >
            <li
              className="content-item"
            >
              <div
                className="content-card"
              />
            </li>
            <li
              className="content-item"
            >
              <div
                className="content-card"
              />
            </li>
          </ul>
        </div>
        <div
          className="content-card padding-bottom-none margin-bottom--half bg-trans"
        >
          <h4
            className="content-card_title t-serif"
          >
            Accessibility, Amenities, Fees
          </h4>
          <ul
            className="content-grid"
          >
            <li
              className="content-item"
            >
              <div
                className="content-card"
              />
            </li>
            <li
              className="content-item"
            >
              <div
                className="content-card"
              />
            </li>
          </ul>
        </div>
        <div
          className="content-card padding-bottom-none margin-bottom--half bg-trans"
        >
          <h4
            className="content-card_title t-serif"
          >
            Lottery Information
          </h4>
          <ul
            className="content-grid"
          >
            <li
              className="content-item"
            >
              <div
                className="content-card"
              >
                <div
                  className="margin-bottom--half"
                >
                  <h4
                    className="t-sans t-small t-bold no-margin"
                  >
                    Lottery Date
                  </h4>
                  <p>
                    04/23/2018
                  </p>
                </div>
                <div
                  className="margin-bottom--half"
                >
                  <h4
                    className="t-sans t-small t-bold no-margin"
                  >
                    Lottery Results Date
                  </h4>
                  <p>
                    04/23/2018
                  </p>
                </div>
              </div>
            </li>
            <li
              className="content-item"
            >
              <div
                className="content-card"
              />
            </li>
          </ul>
        </div>
        <div
          className="content-card padding-bottom-none margin-bottom--half bg-trans"
        >
          <h4
            className="content-card_title t-serif"
          >
            Application Information
          </h4>
          <ul
            className="content-grid"
          >
            <li
              className="content-item"
            >
              <div
                className="content-card"
              />
            </li>
            <li
              className="content-item"
            >
              <div
                className="content-card"
              />
            </li>
          </ul>
        </div>
        <div
          className="content-card padding-bottom-none margin-bottom--half bg-trans"
        >
          <h4
            className="content-card_title t-serif"
          >
            Leasing Agent and Developer Information
          </h4>
          <ul
            className="content-grid"
          >
            <li
              className="content-item"
            >
              <div
                className="content-card"
              />
            </li>
            <li
              className="content-item"
            >
              <div
                className="content-card"
              />
            </li>
          </ul>
        </div>
        <div
          className="content-card padding-bottom-none margin-bottom--half bg-trans"
        >
          <h4
            className="content-card_title t-serif"
          >
            Additional Eligibility Rules
          </h4>
          <ul
            className="content-grid"
          >
            <li
              className="content-item"
            >
              <div
                className="content-card"
              />
            </li>
            <li
              className="content-item"
            >
              <div
                className="content-card"
              />
            </li>
          </ul>
        </div>
        <div
          className="content-card padding-bottom-none margin-bottom--half bg-trans"
        >
          <h4
            className="content-card_title t-serif"
          >
            Additional Information
          </h4>
          <ul
            className="content-grid"
          >
            <li
              className="content-item"
            >
              <div
                className="content-card"
              />
            </li>
            <li
              className="content-item"
            >
              <div
                className="content-card"
              />
            </li>
          </ul>
        </div>
      </div>
    </div>
  </section>,
]
`;
